src/class/hid/hid_device.c

```diff
@@ -202,7 +202,29 @@ bool hidd_control_request(uint8_t rhport, tusb_control_request_t const * p_reque
     uint8_t const desc_index = tu_u16_low (p_request->wValue);
     (void) desc_index;
 
-    if (p_request->bRequest == TUSB_REQ_GET_DESCRIPTOR && desc_type == HID_DESC_TYPE_REPORT)
+    if (p_request->bRequest == TUSB_REQ_GET_DESCRIPTOR && desc_type == HID_DESC_TYPE_HID)
+    {
+      // FIXME: Should check which is the active configuration, but no callback in usbd currently exists to do that

```

Comment:
@pigrew we can have an pointer to store the HID Descriptor in `hidd_open()` at line 169 and used it to return here. That will solve this issue